docs/egress.md

```diff
@@ -79,6 +80,9 @@ spec:
         role: web
   egressIP: 10.10.0.8 # can be populated by Antrea after assigning an IP from the pool below
   externalIPPool: prod-external-ip-pool
+  rateLimit:
+    peak: 300
+    burst: 500
```

Comment:
I rarely see similar bandwidth configurations have peak and burst (and don't both of them mean burst?), the common configuration is just one value, the maximum bandwidth, like Kubernetes bandwidth annotation `kubernetes.io/egress-bandwidth: 10M`, I wonder if we should just have a field `bandwidth: xx`.
And to make the API more straightforward, could we also support different units, like the annotation, 10K, 10M, 10G?
cc @jianjuns @antoninbas @salv-orlando for opinions.

Reply:
I agree that "peak" is a bit confusion, as opposed to "rate" (or maybe "bandwidth"). I also agree that we should support strings instead of integers, as users will be able to provide the unit and it will be more user-friendly.

However, I feel like we should not follow the model of the K8s annotation, which provides the ability to configure the rate without the ability to configure the burst:
- The K8s bandwidth support has been "experimental" for years, so maybe not a great model to follow
- Setting the burst when configuring the CNI [bandwidth plugin](https://www.cni.dev/plugins/current/meta/bandwidth/) / TC is left to the discretion of the container runtime. Containerd sets it to `MaxUint32` (https://github.com/containerd/containerd/blob/main/pkg/cri/server/sandbox_run.go#L548-L556), which _seems_ to translate to 256Mb in TC
- Some users complain about the lack of control over the burst size: https://github.com/kubernetes/kubernetes/issues/90838. This is very limiting for users who want to do traffic shaping

A configuration like this would be nice IMO:
```yaml
bandwidth: #could be called trafficShaping as well
    rate: 10M # 10Mbps
    burst: 32M #32Mb
```

Reply:
@antoninbas's API proposal makes sense to me.

Reply:
Changed.

Reply:
I believe we agreed to use strings with units? The current example is missing units

Reply:
Could we use the term "bandwidth"? My understand is "traffic shaping" is the method while "bandwidth" is the expectation. For a declarative API, I think it's better to declare the desired state, not how it's achieved. Like how cpu and memory is configured in Pod API, it wouldn't mention "cgroup" or "resource management".
```
  containers:
  - name: app
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
```

Reply:
I only mean the API, using "traffic shaping" in the feature gate name and documentation is fine by me.

Reply:
Got you. Changed.